client/app/dedicatedCloud/datacenter/backup/enableAgora/dedicatedCloud-datacenter-backup-enableAgora.controller.js

```diff
@@ -0,0 +1,91 @@
+angular.module('App').controller('DedicatedCloudSubDatacenterVeeamBackupEnableAgoraCtrl', class {
+  constructor($q, $state, $stateParams, $translate, $window,
+    Alerter, DedicatedCloud, OvhHttp, User) {
+    this.$q = $q;
+    this.$state = $state;
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.$window = $window;
+    this.Alerter = Alerter;
+    this.DedicatedCloud = DedicatedCloud;
+    this.OvhHttp = OvhHttp;
+    this.User = User;
+
+    this.loading = {
+      init: false,
+    };
+
+    this.model = {
+      version: 'classic',
+    };
+  }
+
+  closeModal() {
+    this.$state.go('^');
+  }
+
+  onBackupEnableFormSubmit() {
+    if (!this.hosts.length || !this.offer) {
+      return this.closeModal();
+    }
+
+    const productToOrder = {
+      productId: 'privateCloud',
+      serviceName: this.$stateParams.productId,
+      planCode: this.offer.planCode,
+      duration: 'P1M',
+      pricingMode: 'pcc-servicepack-nsx',
+      quantity: 1,
+      configuration: [{
+        label: 'datacenter_id',
+        value: this.datacenter.datacenterId,
+      }, {
+        label: 'offer_type',
+        value: this.model.version,
+      }],
+    };
+
+    return this.User.getUrlOf('express_order')
+      .then((url) => {
+        this.$window.open(`${url}review?products=${JSURL.stringify([productToOrder])}`, '_blank');
+      })
+      .catch((error) => {
+        this.Alerter.error([this.$translate.instant('dedicatedCloud_tab_veeam_enable_fail'), _.get(error, 'data.message')].join(' '));
+        this.closeModal();
```

Comment:
Not necessary if you call it in `finally`

Reply:
indeed :-)